src/routes/roles_routes.rs

```diff
@@ -0,0 +1,270 @@
+    path = "/user/{user_id}/tournament/{tournament_id}/roles",
+    responses(
+        (
+        status=200, description = "Roles created successfully",
+        body=Vec<Role>
+        ),
+        (status=400, description = "Bad request"),
+        (
+            status=401,
+            description = "The user is not permitted to modify roles within this tournament"
+        ),
+        (status=404, description = "User of tournament not found"),
+        (status=409, description = "The user is already granted roles within this tournament. Use PATCH method to modify user roles"),
+        (status=500, description = "Internal server error"),
+    ),
+    tag = "roles"
+)]
+async fn create_user_roles(
+    State(state): State<AppState>,
+    headers: HeaderMap,
+    cookies: Cookies,
+    Path((user_id, tournament_id)): Path<(Uuid, Uuid)>,
+    Json(json): Json<Vec<Role>>,
+) -> Result<Response, OmniError> {
+    let pool = &state.connection_pool;
+    let tournament_user =
+        TournamentUser::authenticate(tournament_id, &headers, cookies, pool).await?;
+
+    match tournament_user.has_permission(Permission::WriteRoles) {
+        true => (),
+        false => return Err(OmniError::UnauthorizedError),
+    }
+
+    let target_user = User::get_by_id(user_id, pool).await?;
+    let roles = target_user.get_roles(tournament_id, pool).await?;
+    if !roles.is_empty() {
+        return Err(OmniError::ResourceAlreadyExistsError);
+    }
+
+    match Role::post(user_id, tournament_id, json, pool).await {
+        Ok(role) => Ok(Json(role).into_response()),
+        Err(e) => {
+            error!(
+                "Error creating roles for user {} within tournament {}: {e}",
+                user_id, tournament_id
+            );
+            Err(e)?
+        }
+    }
+}
+
+/// List roles a user is given within a tournament
+///
+/// The user must be given a role within this tournament to use this endpoint.
+#[utoipa::path(get, path = "/user/{user_id}/tournament/{tournament_id}/roles",
+    responses(
+        (status=200, description = "Ok", body=Vec<Role>,
+            example=json!(get_roles_example())
+        ),
+        (status=400, description="Bad request"),
+        (status=401, description="The user is not permitted to see roles, meaning they don't have any role within this tournament"),
+        (status=404, description="User or tournament not found"),
+        (status=500, description="Internal server error"),
+    ),
+    tag = "roles"
+)]
+async fn get_user_roles(
+    State(state): State<AppState>,
+    headers: HeaderMap,
+    cookies: Cookies,
+    Path((user_id, tournament_id)): Path<(Uuid, Uuid)>,
+) -> Result<Response, OmniError> {
+    let pool = &state.connection_pool;
+    let tournament_user =
+        TournamentUser::authenticate(tournament_id, &headers, cookies, &pool).await?;
+
+    match tournament_user.roles.is_empty() {
+        true => (),
+        false => return Err(OmniError::UnauthorizedError),
+    }
```

Comment:
I don't think that was what you meant to do here.
```suggestion
    if tournament_user.roles.is_empty() {
        return Err(OmniError::UnauthorizedError);
    }
```